op/model.py

```diff
@@ -48,31 +49,58 @@ def get_unit(self, unit_name):
 
 
 class ModelCache(weakref.WeakValueDictionary):
+
+    def __init__(self, *args, **kw):
+        self.backend = kw.pop('backend')
+        super().__init__(*args, **kw)
+
     def get(self, entity_type, *args):
         key = (entity_type,) + args
         entity = super().get(key)
         if entity is None:
-            entity = entity_type(*args, cache=self)
+            entity = entity_type(*args, backend=self.backend, cache=self)
             self[key] = entity
         return entity
 
 
 class Application:
-    def __init__(self, name, cache):
+    def __init__(self, name, backend, cache):
         self.name = name
+        self._is_local = None
```

Comment:
self.is_local = (name == backend.local_app_name)

Reply:
The problem with that is that it will always be done on object creation which will force modification of test_charm.py - I will have to provide a fake JUJU_UNIT_NAME in the `setUp` code.